pkg/verifier/vulnerability/ibm_va.go

```diff
@@ -0,0 +1,264 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vulnerability
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	ibmcloud "github.com/IBM/go-sdk-core/v4/core"
+	"github.com/IBM/portieris/helpers/credential"
+	"github.com/IBM/portieris/helpers/image"
+	"github.com/golang/glog"
+)
+
+const maxRetries = 3
+
+var sleepTime = time.Second
+
+// IBMVAScanner is a client wrapper for interacting with Vulnerability Advisor for IBM Cloud Container Registry API
+type IBMVAScanner struct {
+	credentials          credential.Credentials
+	client               HTTPClient
+	Timeout              int64
+	AccountHeader        string
+	authenticatorFactory authenticatorFactoryInterface
+}
+
+// HTTPClient makes testing IBMVAScanner simpler
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+var (
+	// ErrorUnauthorised is a typed error for unauthorised requests
+	ErrorUnauthorised = errors.New("unauthorised")
+)
+
+// NewIBMVulnerabilityAdvisorScanner returns a new client for IBM's Vulnerability Advisor
+func NewIBMVulnerabilityAdvisorScanner(credentials credential.Credentials, account string) *IBMVAScanner {
+	return &IBMVAScanner{
+		credentials:   credentials,
+		AccountHeader: account,
+		client: &http.Client{
+			Timeout: time.Second * time.Duration(10),
+		},
+		authenticatorFactory: authenticatorFactory{},
+	}
+}
+
+// IBMVASummary represents the summary datatype returned by the VA API
+type IBMVASummary struct {
+	Status                        string `json:"status" description:"Overall vulnerability assessment status from: OK, WARN, BLOCK, UNSUPPORTED, INCOMPLETE, UNSCANNED"`
+	ScanTime                      int64  `json:"scan_time" description:"The scan time of the report as a UNIX timestamp"`
+	IssueCount                    int    `json:"issue_count" description:"The number of issues found"`
+	ExemptIssueCount              int    `json:"exempt_issue_count" description:"The number of exempt issues found"`
+	VulnerabilityCount            int    `json:"vulnerability_count" description:"The number of vulnerability issues found"`
+	ExemptVulnerabilityCount      int    `json:"exempt_vulnerability_count" description:"The number of exempt vulnerability issues found"`
+	ConfigurationIssueCount       int    `json:"configuration_issue_count" description:"The number of configuration issues found"`
+	ExemptConfigurationIssueCount int    `json:"exempt_configuration_issue_count" description:"The number of exempt configuration issues found"`
+}
+
+// CanImageDeployBasedOnVulnerabilities is an implementation of the Scanner interface for IBM VA
+func (s *IBMVAScanner) CanImageDeployBasedOnVulnerabilities(image image.Reference) (scan ScanResponse, err error) {
+	if !image.HasIBMRepo() {
+		return scan, fmt.Errorf("Cannot use IBM Vulnerability Advisor with image %q", image.String())
```

Comment:
So these logs are why I commented on the last PR saying could that say `vulnerability:` at the front... could these have `IBMVA:` on the front?

Also - I think these logs need to use the product names "Vulnerability Advisor" or "Vulnerability Advisor for IBM Cloud Container Registry" instead of "IBM VA" or "IBM Vulnerability Advisor".